MariiaNovitska/HW10_OOP/task-1.py

```diff
@@ -0,0 +1,48 @@
+class Polygone():
+   
+    def __init__(self, sides):
+        self.side_number = sides
+        self.sides = [float(input(f"Enter {x+1} side: ")) for x in range(sides)]
+
+
+    def show_sides(self):
+        counter = 1
+        for x in self.sides:            
+            print (f"Side {counter} = {x}")
+            counter +=1
+
+
+class Rectangle(Polygone):
+
+    def __init__(self):
+        super().__init__(2)
+
+
+    def show_sides(self):
+        counter = 1
+        for x in self.sides*2:            
+            print (f"Side {counter} = {x}")
+            counter +=1
+
+
+    def area (self):
+        s = self.sides[0]*self.sides[1]
+        return s
+    
+
+# Створила клас Трикутник з перевіркою на існування трикутнику
```

Comment:
good job
but look at this implementation

class Polygon:
    def __init__(self, sides):
        self.side_number = sides
        self.sides = [float(input(f"Enter side {x+1}: ")) for x in range(sides)]

    def show_sides(self):
        for i, side in enumerate(self.sides, start=1):
            print(f"Side {i} = {side}")


class Triangle(Polygon):
    def __init__(self):
        super().__init__(3)
        
    def calculate_area(self):
        a, b, c = self.sides
        sorted_sides = sorted(self.sides)
        if sorted_sides[0] + sorted_sides[1] > sorted_sides[2]:
            s = (a + b + c) / 2
            area = (s * (s - a) * (s - b) * (s - c)) ** 0.5
            return area
        else:
            print("Cannot calculate area. The triangle does not exist.")


tri = Triangle()
area = tri.calculate_area()
if area is not None:
    print("Triangle area:", round(area,2))